### PR TITLE
Add back `PyBytes::new`

### PR DESCRIPTION
**Pull request by bschoenmaeckers**
Add back the `PyBytes` constructors without `_bound`.

### CODE REVIEW

**Comment by alex** [comment]
Right now you've entirely removed the `_bound` names. Those need to be kept around for backwards compatibility.

**Comment by bschoenmaeckers** [comment]
> Right now you've entirely removed the `_bound` names. Those need to be kept around for backwards compatibility.

Oops forgot that, will at them back shortly!

File: src/types/bytes.rs [hunk 1]
```diff
@@ -116,11 +133,18 @@ impl PyBytes {
...
+    pub unsafe fn from_ptr(py: Python<'_>, ptr: *const u8, len: usize) -> Bound<'_, PyBytes> {
         ffi::PyBytes_FromStringAndSize(ptr.cast(), len as isize)
             .assume_owned(py)
             .downcast_into_unchecked()
     }
+
+    /// Deprecated name for [`PyBytes::from_ptr`].
+    #[deprecated(since = "0.23.0", note = "renamed to `PyBytes::from_ptr`")]
+    #[inline]
+    pub unsafe fn from_ptr_bound(py: Python<'_>, ptr: *const u8, len: usize) -> Bound<'_, PyBytes> {
```
Comment (LilyFoote) [inline src/types/bytes.rs:145]:
```suggestion
    pub unsafe fn bound_from_ptr(py: Python<'_>, ptr: *const u8, len: usize) -> Bound<'_, PyBytes> {
```

**Review by LilyFoote — CHANGES_REQUESTED** [review]

File: src/types/bytes.rs (hunk 1 above)
Comment (bschoenmaeckers) [reply src/types/bytes.rs]: ~3th~ 4th time's a charm

File: src/types/bytes.rs [hunk 2]
```diff
@@ -116,11 +133,18 @@ impl PyBytes {
...
     /// `std::slice::from_raw_parts`, this is
     /// unsafe](https://doc.rust-lang.org/std/slice/fn.from_raw_parts.html#safety).
-    pub unsafe fn bound_from_ptr(py: Python<'_>, ptr: *const u8, len: usize) -> Bound<'_, PyBytes> {
+    pub unsafe fn from_ptr(py: Python<'_>, ptr: *const u8, len: usize) -> Bound<'_, PyBytes> {
         ffi::PyBytes_FromStringAndSize(ptr.cast(), len as isize)
             .assume_owned(py)
             .downcast_into_unchecked()
     }
+
+    /// Deprecated name for [`PyBytes::from_ptr`].
```
Comment (Icxolu) [inline src/types/bytes.rs:142]: We need to include the `Safety` section from `from_ptr` here as well to make CI green.

**Review by LilyFoote — APPROVED** [review]

File: src/types/bytes.rs (hunk 2 above)
Comment (bschoenmaeckers) [reply src/types/bytes.rs:142]: Added the safety note.

**Review by Icxolu — APPROVED** [review]

**Pull request merged**